src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesImplementation.cs

```diff
@@ -34,28 +69,39 @@ public sealed override void GenerateIV()
 
         public sealed override void GenerateKey()
         {
-            Key = RandomNumberGenerator.GetBytes(KeySize / BitsPerByte);
+            Span<byte> key = stackalloc byte[KeySize / BitsPerByte];
+            RandomNumberGenerator.Fill(key);
+            SetKey(key);
         }
 
         protected sealed override void Dispose(bool disposing)
         {
             base.Dispose(disposing);
         }
 
+        protected override void SetKeyCore(ReadOnlySpan<byte> key)
+        {
+            KeySizeValue = checked(BitsPerByte * key.Length);
+            _keyBox?.Dispose();
+            _keyBox = new FixedMemoryKeyBox(key);
```

Comment:
It'd be dangerous to reuse an existing box?

Reply:
The boxes might be different sizes, for one. But I generally get "bad vibes" from updating a key in-place instead of swapping out references.

Reply:
Yeah, my thoughts mirror Kevin's, that it's just a sort of heebie-jeebie.  The expected case is that SetKey (or set_Key) is only called once, so it doesn't matter greatly.

If we wanted, we could always allocate the box to the maximum keysize and hold the span down to the "in use" portion.  Parallel use (not supported) may do weird things, but since all memory reads are contained within the UseKey callback scope (and an AddRef/Release pair) it won't AV.  That copy/update would have to be done inside an AddRef/Release scope, of course, so we're not writing during a parallel call to Dispose...  an interesting thought experiment, and maybe something to come back to, but I don't think it needs to be done right now (given the low expected savings)

Reply:
Ok